server/src/main/java/org/apache/cassandra/sidecar/acl/AuthCache.java

```diff
@@ -61,14 +63,16 @@ protected AuthCache(String name,
                         ExecutorPools executorPools,
                         Function<K, V> loadFunction,
                         Supplier<Map<K, V>> bulkLoadFunction,
-                        CacheConfiguration cacheConfiguration)
+                        CacheConfiguration cacheConfiguration,
+                        CacheStatsCounter cacheMetrics)
     {
         this.name = name;
         this.vertx = vertx;
         this.internalPool = executorPools.internal();
         this.loadFunction = loadFunction;
         this.bulkLoadFunction = bulkLoadFunction;
         this.config = cacheConfiguration;
+        this.cacheMetrics = cacheMetrics;
```

Comment:
we should validate this is not null
```suggestion
        this.cacheMetrics = Objects.requireNonNull(cacheMetrics, "cacheMetrics is required");
```